worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -509,6 +513,50 @@ private[celeborn] class Worker(
     // If worker register still failed after retry, throw exception to stop worker process
     throw new CelebornException("Register worker failed.", exception)
   }
+
+  private def handleResourceConsumption(): util.Map[UserIdentifier, ResourceConsumption] = {
+    val resourceConsumptionSnapshot = storageManager.userResourceConsumptionSnapshot()
+    resourceConsumptionSnapshot.foreach(resourceConsumption => {
+      resourceConsumptionSource.addGauge(
+        ResourceConsumptionSource.DISK_FILE_COUNT,
+        resourceConsumption._1.toMap) { () =>
+        computeUserResourceConsumption(resourceConsumption).diskFileCount
+      }
+      resourceConsumptionSource.addGauge(
+        ResourceConsumptionSource.DISK_BYTES_WRITTEN,
+        resourceConsumption._1.toMap) { () =>
+        computeUserResourceConsumption(resourceConsumption).diskBytesWritten
+      }
```

Comment:
Seems we should not register metrics here?

Reply:
@AngersZhuuuu, due to invoke `addGauge[T](name: String, labels: Map[String, String] = Map.empty)(f: () => T)`, the metrics would be registered for new userIdentifier.